tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1389,6 +1389,12 @@ def eval_fn(m):
 
 def create_value_transform(echo_nostr, echo_noarr):
     def value_transform(val):
+        if not isinstance(val, genpy.Message):
+            if echo_nostr and isinstance(val, str):
+                return None
+            elif echo_noarr and isinstance(val, list):
+                return None
```

Comment:
Here I don't see how we could print the type information:

`return '<array length: %s>' % len(val)`